tests/Workleap.DotNet.CodingStandards.Tests/PackageFixture.cs

```diff
@@ -0,0 +1,56 @@
+using System.Diagnostics;
+using CliWrap;
+using Meziantou.Framework;
+using Workleap.DotNet.CodingStandards.Tests.Helpers;
+
+namespace Workleap.DotNet.CodingStandards.Tests;
+
+public sealed class PackageFixture : IAsyncLifetime
+{
+    private readonly TemporaryDirectory _packageDirectory = TemporaryDirectory.Create();
+
+    public FullPath PackageDirectory => _packageDirectory.FullPath;
+
+    public async Task InitializeAsync()
+    {
+        var nuspecPath = PathHelpers.GetRootDirectory() / "Workleap.DotNet.CodingStandards.nuspec";
+        string[] args = ["pack", nuspecPath, "-ForceEnglishOutput", "-Version", "999.9.9", "-OutputDirectory", _packageDirectory.FullPath];
+
+        if (OperatingSystem.IsWindows())
+        {
+            var exe = FullPath.GetTempPath() / $"nuget-{Guid.NewGuid()}.exe";
+            await DownloadFileAsync("https://dist.nuget.org/win-x86-commandline/latest/nuget.exe", exe);
+
+            await Cli.Wrap(exe)
+                .WithArguments(args)
+                .ExecuteAsync();
+        }
+        else
+        {
+            // CliWrap doesn't support UseShellExecute. On Linux, it's easier to use it as "nuget" is a shell script that use mono to run nuget.exe
```

Comment:
I believe that's because we don't consume the stdout and stderr of the process.

If we assume that nuget is available in the PATH (maybe it should be explained in the README), then I think we could get rid of the OS check, the nuget.exe download and simply execute this code which should be cross platform:

```csharp
await Cli.Wrap(exe)
    .WithArguments(args)
    .ExecuteBufferedAsync(); // <-- "using CliWrap.Buffered;"
```

I experienced the same issue when executing `git` and `docker` using CliWrap on Linux.

Reply:
Not having to download NuGet.exe would mean that we could get rid of the HttpClient specific code!

Reply:
CliWrap explicitly uses `UseShellExecute = false`, and I don't think there is a way to override this behavior (https://github.com/Tyrrrz/CliWrap/blob/1cfa50eaaf8b8c74522a999e89dcf3577602eae3/CliWrap/Command.Execution.cs#L83). On the runner (Linux), nuget is actually a script (`/bin/sh` that calls `mono nuget.exe <args>`). `nuget.exe` is only available on Windows, and there is no version for Linux or MacOS, that's why it relies on mono.

To use CliWrap I think you need to wrap `/bin/sh -c "encoded parameters"`.

Requiring `nuget` to be in the path increase the friction for devs. As most devs, in our case, are on Windows, I'm fine with the solution that is more automated on Windows than on Linux.

Reply:
I know their implementation and this mono nuget.exe thing, but the GitHub CI runs just fine with this code on ubuntu-latest (I forked this branch). It also works on Windows.

![image](https://github.com/gsoft-inc/wl-dotnet-codingstandards/assets/14242083/03f0a7e1-6a4b-4717-a25c-0f226ca79ecb)

![image](https://github.com/gsoft-inc/wl-dotnet-codingstandards/assets/14242083/67d2af8c-663c-4cd0-912b-c77279508ad7)

Reply:
You are right! I'm not sure what I did test yesterday evening...